src/senselab/audio/data_structures/audio.py

```diff
@@ -83,9 +108,82 @@ def from_filepath(cls, filepath: str | os.PathLike, metadata: Dict = {}) -> "Aud
         if not os.path.exists(filepath):
             raise FileNotFoundError(f"File {filepath} does not exist.")
 
-        array, sampling_rate = torchaudio.load(filepath)
+        if offset_in_sec < 0:
+            raise ValueError("Offset must be a non-negative value.")
+        if duration_in_sec < -1:
+            raise ValueError("Duration must be -1 (to read full file) or a positive value.")
+
+        # Retrieve audio metadata
+        try:
+            info = torchaudio.info(filepath)
+            sampling_rate = info.sample_rate
+            total_frames = info.num_frames
+        except Exception as e:
+            raise ValueError(f"Failed to retrieve audio metadata for file {filepath}: {str(e)}") from e
+
+        # Convert time-based offset and duration to frame-based values
+        frame_offset = int(offset_in_sec * sampling_rate)
+        if frame_offset > total_frames:
+            raise ValueError(
+                f"Offset ({offset_in_sec} seconds) exceeds the duration of the audio file ({filepath}): "
+                f"{total_frames / sampling_rate} seconds."
+            )
+
+        num_frames = int(duration_in_sec * sampling_rate) if duration_in_sec > 0 else -1
+
+        # Ensure num_frames are within bounds
+        if num_frames > 0:
```

Comment:
Technically this if statement is not necessary though it is useful for readability, if num_frames is not greater than 0 then the min will still return -1